packages/store-sync/src/common.ts

```diff
@@ -119,5 +119,5 @@ export type SyncResult<TConfig extends StoreConfig = StoreConfig> = {
   latestBlockNumber$: Observable<bigint>;
   blockLogs$: Observable<BlockLogs>;
   blockStorageOperations$: Observable<BlockStorageOperations<TConfig>>;
-  waitForTransaction: (tx: Hex) => Promise<{ receipt: TransactionReceipt }>;
+  waitForTransaction: (tx: Hex) => Promise<void>;
```

Comment:
Sometimes the tx is found via the processed storage operations, thus I don't have a transaction receipt, so we can't return one here. If downstream usage needs one, they can always do something like:

```ts
await waitForTransaction(tx);
const receipt = publicClient.getTransactionReceipt({ hash: tx });
```